### PR TITLE
componentizied input-search and search in set currency and token management

### PR DESCRIPTION
**Pull request by tbenr**
- added search feature when selecting default currency
- added search feature in asset management
- moved search-input to a dedicated ui component
- removed auto-correction and auto-capitalization in search input

image

image

**note for testing:** non-regression tests should be done on chat search:
image

status: ready

### CODE REVIEW

**Comment by tbenr** [comment]
@rachelhamlin to set default currency as per device settings we need to move from `react-native-languages` to `react-native-localize` dependency, which takes more time. For now I implemented better way to set it :)

**Comment by flexsurfer** [comment]
thanks @tbenr ! that's great! do you think  we could add search for assets management in this PR or in a separate one  ?

**Comment by tbenr** [comment]
@flexsurfer this morning I realize it was missing there too :) let me add it in this pr

**Comment by tbenr** [comment]
@flexsurfer done!

File: src/status_im/search/core.cljs [hunk 1]
```diff
@@ -1,5 +1,17 @@
 (ns status-im.search.core
   (:require [status-im.utils.fx :as fx]))
 
-(fx/defn filter-changed [cofx search-filter]
+(fx/defn filter-changed
```
Comment (tbenr) [inline src/status_im/search/core.cljs]: @yenda maybe here and in other places could make sense to rename `filter-changed` and all other related events\keys to `chat-filtered-changed`. So we have consistency across all search filter stuff. WDYT?

**Comment by errorists** [comment]
@tbenr thanks man! I wouldn't have it focused on entering the view tho. It's not critical but if possible I'd ask to change that

**Review by errorists — APPROVED** [review]

**Comment by tbenr** [comment]
@errorists oky no prob!

File: src/status_im/search/core.cljs (hunk 1 above)
Comment (tbenr) [reply src/status_im/search/core.cljs]: @yenda i did that :)

**Comment by tbenr** [comment]
@errorists done!

**Review by flexsurfer — APPROVED** [review]

**Comment by tbenr** [comment]
rebased

**Comment by churik** [comment]
Thank you for contribution and apologies for the delay in testing.
Noticed one small issue: value in the search field in asset or currency management is erased while switching tabs.
It is not reproducible with the search field in chat.
To reproduce type some value in search in wallet > settings > manage assets, switch tab and go back to wallet.
Expected, that value will be preserved.
In fact, value is empty, but search results are filtered according to the previous query.
manage-asset

**Comment by tbenr** [comment]
@churik ok yes i recreated it. give me some time to fix it, i'm pretty busy these days.

**Comment by tbenr** [comment]
@churik should be fixed now, I rebased too.

**Comment by churik** [comment]
Retested all search fields on IOS13 and Android8.
Looks good to me! 
Test case is added https://ethstatus.testrail.net/index.php?/cases/view/6269

**Pull request merged**